cdviz-collector/src/main.rs

```diff
@@ -41,15 +43,27 @@ type Receiver<T> = tokio::sync::broadcast::Receiver<T>;
 #[derive(Clone, Debug)]
 struct Message {
     received_at: OffsetDateTime,
-    cdevent: CDEvent,
+    cd_event: Option<CDEvent>,
+    cloud_event: Option<Event>,
     //raw: serde_json::Value,
```

Comment:
This kind of structure is not what you want, because it allows invalid state like `Some,Some` & `None,None`. But what (I image) you want to express is cdevents XOR cloudevent. In this case you should use enum.

Anyway for Message I would like to avoid the duality. So if sink (consumer of the message) needs info from the cloudevent part then replace `cdevent` by `cloudevent`. Else keep Message as it is (my favorite, until consumer need cloudevent metadata, and maybe in this case, I like to have the `CDEvent` type enforced by Message, and that is not the case with Event).